pkg/apis/configuration/v1/kongingress_types.go

```diff
@@ -34,9 +34,15 @@ type KongIngress struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// Upstream represents a virtual hostname and can be used to loadbalance
+	// incoming requests over multiple services (targets).
```

Comment:
The use of the words "services" in this sense is a bit overloaded (unfortunately we have kong services, kubernetes services, and then the word "services" itself being a bit overloaded), perhaps:

```suggestion
	// Upstream represents a virtual hostname and can be used to loadbalance
	// incoming requests over multiple targets (e.g. Kubernetes `Services` can
	// be a target, OR `Endpoints` can be targets).
```